tensorflow_probability/python/mcmc/kernel.py

```diff
@@ -84,3 +84,22 @@ def bootstrap_results(self, init_state):  # pylint: disable=unused-argument
         `Tensor`s representing internal calculations made within this function.
     """
     return []
+
+  def copy(self, **override_parameter_kwargs):
+    """Non-destructively creates a deep copy of the kernel
+
+    Args:
+      override_parameter_kwargs: Python String/value `dictionary` of
+        initialization arguments to override with new values.
+
+    Returns:
+      new_kernel: `TransitionKernel` object of same type as `self`,
+        initialized with the union of self.parameters and
+        override_parameter_kwargs, with any shared keys overridden by the
+        value of override_parameter_kwargs, i.e.,
+        `dict(self.parameters, **override_parameters_kwargs)`.
+    """
+    parameters = dict(self.parameters, **override_parameter_kwargs)
+    new_kernel = type(self)(**parameters)
+    new_kernel._parameters = parameters # pylint: disable=protected-access
```

Comment:
is this needed? doesn't the __init__ call that creates new_kernel set them?

Reply:
May be redundant now that I look at it; the copy method in `Distribution` has a similar pattern.